crates/solidity/src/solc/standard_json/output/mod.rs

```diff
@@ -58,21 +58,51 @@ impl Output {
         solc_version: &SolcVersion,
         debug_config: &revive_llvm_context::DebugConfig,
     ) -> anyhow::Result<Project> {
+        trace!("Starting try_to_project");
+
         if let SolcPipeline::EVMLA = pipeline {
             self.preprocess_dependencies()?;
         }
 
         let files = match self.contracts.as_ref() {
-            Some(files) => files,
-            None => {
-                anyhow::bail!(
-                    "{}",
-                    self.errors
-                        .as_ref()
-                        .map(|errors| serde_json::to_string_pretty(errors).expect("Always valid"))
-                        .unwrap_or_else(|| "Unknown project assembling error".to_owned())
-                );
+            Some(files) => {
+                trace!("Found contracts: {:#?}", files.keys());
+                files
             }
+            None => &{
+                match &self.errors {
+                    Some(errors) if !errors.is_empty() => {
+                        // Here we want to ensure that we check for actual errors
+                        // Warnings are valid and shouldnt be treated as regualar errors
+                        // As such we need to
+                        //1. Filter errors for severity "error"
+                        //2. If we do have errors only then do we bail
+                        //3. If we dont have errors we simply return empty files
+                        let has_errors = errors.iter().any(|e| e.severity == "error");
+
+                        if has_errors {
+                            error!("Compiler errors found: {:#?}", errors);
+                            anyhow::bail!(
+                                "{}",
+                                serde_json::to_string_pretty(errors).expect("Always valid")
+                            );
+                        }
+
+                        warn!("Compiler warnings found: {:#?}", errors);
```

Comment:
Useless log statements as everything should already be displayed via the compiler warnings and errors.